src/components/edit/EditGlStyleDirective.js

```diff
@@ -20,23 +20,49 @@ goog.provide('ga_editglstyle_directive');
       link: function(scope, element, attrs, controller) {
 
         scope.$watch('glStyle', function(newGlStyle) {
-          if (!newGlStyle || !scope.config || !scope.config.selectableLayers) {
+          if (!newGlStyle || !scope.config || !scope.config.length) {
             return;
           }
 
-          scope.selectableLayers = [];
+          /**
+           * Object containing list of glStyle layers to modfiy for each value
+           * in config.selectableLayers.
+           */
+          scope.groups = [];
 
           scope.glStyle.layers.forEach(function(layer) {
-            if (scope.config.selectableLayers.indexOf(layer.id) !== -1) {
-              scope.selectableLayers.push(layer);
-            }
+            scope.config.forEach(function(edit) {
+              var regex = new RegExp(edit.regex || edit.id);
+              if (regex.test(layer.id) || regex.test(layer['source-layer'])) {
+                if (!scope.groups[edit.id]) {
+                  scope.groups[edit.id] = [];
+                }
+                scope.groups[edit.id].push(layer);
+              }
+            });
           });
-          scope.selectedLayer = scope.selectableLayers[0];
+          scope.edit = scope.config[0];
+          scope.group = scope.groups[scope.edit.id];
         });
 
-        scope.useWidget = function(widget, e) {
+        scope.useWidget = function(widget, path) {
           var regex = new RegExp(widget);
-          return regex.test(e[2]);
+          return regex.test(path[2]);
+        }
```

Comment:
semi-colon always missing for scope functions is that on purpose?

Reply:
No, maybe my linter.